mycore-ocfl/src/main/java/org/mycore/ocfl/MCROCFLXMLClassificationManager.java

```diff
@@ -0,0 +1,213 @@
+            });
+        } catch (IOException e) {
+            throw new MCRPersistenceException("Failed to update object '" + objName + "'", e);
+        } catch (IllegalArgumentException e) {
+            // Issue seems to be with the OCFL Library trying to find a file it deleted itself
+            // LOGGER.error("Something has gone Wrong!", e);
+        }
+
+    }
+
+    public void fileDelete(MCRCategoryID mcrid) {
+        String objName = getName(mcrid);
+        Date lastModified = new Date(MCRCategoryDAOFactory.getInstance().getLastModified(mcrid.getRootID()));
+        VersionInfo versionInfo = buildVersionInfo(MESSAGE_DELETED, lastModified);
+        getRepository().updateObject(ObjectVersionId.head(objName), versionInfo, updater -> {
+            updater.removeFile(buildFilePath(mcrid));
+        });
+    }
+
+    /**
+     * Load a Classification from the OCFL Store.
+     * @param mcrid ID of the Category
+     * @param revision Revision of the Category or <code>null</code> for HEAD
+     * @return Content of the Classification
+     */
+    public MCRContent retrieveContent(MCRCategoryID mcrid, String revision) {
+        String objName = getName(mcrid);
+        OcflRepository repo = getRepository();
+        ObjectVersionId vId = revision != null ? ObjectVersionId.version(objName, revision)
+            : ObjectVersionId.head(objName);
+
+        try {
+            repo.getObject(vId);
+        } catch (NotFoundException e) {
+            throw new MCRUsageException("Object '" + objName + "' could not be found", e);
+        }
+
+        if (convertMessageToType(repo.getObject(vId).getVersionInfo().getMessage()) == MCROCFLMetadataVersion.DELETED) {
+            throw new MCRUsageException("Cannot read already deleted object '" + objName + "'");
+        }
+
+        try (InputStream storedContentStream = repo.getObject(vId).getFile(buildFilePath(mcrid)).getStream()) {
+            Document xml = new MCRStreamContent(storedContentStream).asXML();
+            if (revision != null) {
+                xml.getRootElement().setAttribute("rev", revision);
+            }
+            return new MCRJDOMContent(xml);
+        } catch (JDOMException | SAXException | IOException e) {
+            throw new MCRPersistenceException("Can not parse XML from OCFL-Store", e);
+        }
+    }
+
+    protected String getName(MCRCategoryID mcrid) {
+        return CLASSIFICATION_PREFIX + mcrid.getRootID();
+    }
+
+    /**
+     * Build file path from ID, use for classifications only!
+     * <p><b>Use {@link #buildFilePath(MCRCategory)} when possible instead</b></p>
+     * @param mcrid The ID to the Classification
+     * @return The Path to the File.
+     */
+    protected String buildFilePath(MCRCategoryID mcrid) {
+        if (mcrid.isRootID()) {
+            return rootFolder + mcrid.toString() + ".xml";
+        } else {
+            return rootFolder + mcrid.getRootID() + '/' + mcrid.toString() + ".xml";
+        }
+    }
```

Comment:
Where are all these files come from? Where is this specified? How to we ensure that if a classification is deleted all categories are deleted, too?